packages/ipfs-core/src/components/files/stat.js

```diff
@@ -77,77 +77,78 @@ const statters = {
     }
   },
   /**
-   * @param {any} file
+   * @param {import('ipfs-unixfs-exporter').UnixFSFile} file
    * @returns {Stat}
    */
-  'dag-pb': (file) => {
-    const blocks = file.node.Links.length
-    const size = file.node.size
-    const cumulativeSize = file.node.size
-
+  file: (file) => {
     /** @type {Stat} */
-    const output = {
+    const stat = {
       cid: file.cid,
       type: 'file',
-      size: size,
-      cumulativeSize: cumulativeSize,
-      blocks: blocks,
+      size: file.unixfs.fileSize(),
+      cumulativeSize: file.node.size,
+      blocks: file.unixfs.blockSizes.length,
       local: undefined,
       sizeLocal: undefined,
-      withLocality: false
+      withLocality: false,
+      mode: file.unixfs.mode
     }
 
-    if (file.unixfs) {
```

Comment:
the things I think are missing from this deleted block now are - `stat.type` setting (it's always `'file'` now), `stat.size` set to `0` for `isDirectory()`, should these be added back?

Reply:
... or maybe this is now safely disambiguating the file vs directory case before we even get here ... I guess that makes them unnecessary here!